examples/wallet/app/components/RestoreWalletFromPrivateKey.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React, { useState } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import typeof { setAccount as SetAccount } from '../actions/account';
+
+type Props = {
+  setAccount: SetAccount
+};
+
+// FIXME: this has no error handling, neither while parsing the address
+// nor when fetching the balance.
+export default ({ setAccount }: Props) => {
+  const handleSubmitCreateSpending = function handleSubmitCreateSpending(
+    event
+  ) {
+    event.preventDefault();
+    if (mustCreateSpendingPassword) {
+      if (checkValidSpendingPassword(password, confirmPassword)) {
+        setAccount(newPrivateKey, password);
+      }
+    } else {
+      setAccount(newPrivateKey, '');
+    }
+  };
+
+  const handleCheckCreateSpendingPassword = function handleCheckCreateSpendingPassword(
+    evt
+  ) {
+    setMustCreateSpendingPassword(evt.target.checked);
+    setHiddenSpendingPassword(!evt.target.checked);
+  };
+
+  const checkValidSpendingPassword = function checkValidSpendingPassword(
+    pass,
+    confirmation
+  ) {
+    if (!pass && !confirmation) return true;
+    if (pass.length < 8) {
+      setIsValidPassword(false);
+      return false;
+    }
+    setIsValidPassword(true);
+
+    if (pass !== confirmation) {
+      setArePasswordAndConfirmationEqual(false);
+      return false;
+    }
+    setArePasswordAndConfirmationEqual(true);
+    return true;
+  };
+
+  const [isValidPassword, setIsValidPassword] = useState(true);
+  const [newPrivateKey, setNewPrivateKey] = useState('');
+  const [mustCreateSpendingPassword, setMustCreateSpendingPassword] = useState(
+    true
+  );
+  const [hiddenSpendingPassword, setHiddenSpendingPassword] = useState(false);
+
+  const [
```

Comment:
I would'nt bother creating a new state variable for something which can be easily computed from existing ones.

Reply:
Deleted

Reply:
hiddenSpendingPassword and setHiddenSpendingPassword does not exists